x/mongo/driver/topology/CMAP_spec_test.go

```diff
@@ -370,20 +437,43 @@ func runOperationInThread(t *testing.T, operation map[string]interface{}, testIn
 			}
 		}
 	case "waitForEvent":
-		targetCount, ok := operation["count"]
-		if !ok {
-			t.Fatalf("count is required to waitForEvent")
+		var targetCount int
+		{
+			f, ok := operation["count"].(float64)
+			if !ok {
+				t.Fatalf("count is required to waitForEvent")
+			}
+			targetCount = int(f)
 		}
-		targetEventName, ok := operation["event"]
+
+		targetEventName, ok := operation["event"].(string)
 		if !ok {
 			t.Fatalf("event is require to waitForEvent")
 		}
 
+		// If there is a timeout specified in the "waitForEvent" operation, then use that timeout.
+		// Otherwise, use a default timeout of 10s when waiting for events. Using a default timeout
+		// prevent the Go test runner from timing out, which just prints a stack trace and no
```

Comment:
```suggestion
		// prevents the Go test runner from timing out, which just prints a stack trace and no
```
Ah great addition, this will help with debugging.